# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLDialogElement.h
# Commit: 130bfa532753
# Full Hash: 130bfa5327532a2b188e4cc06f5e1de7183327e8
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-04-30 09:28:29
# Regressor Bug: 1660271
# File Overlap Count: 1
# Description:
#   Bug 1660271 - Move the focus to the previously focused element when <dialog> is closed r=smaug
#   
#   This change is based on the latest changes for the <dialog> element in
#   https://github.com/whatwg/html/pull/6531, such that closing
#   <dialog> should move the focus to the previously focused element.
# ==============================================================================

diff -r f2ab512468e9 -r 130bfa532753 dom/html/HTMLDialogElement.h
--- a/dom/html/HTMLDialogElement.h	Thu Apr 29 19:33:02 2021 +0000
+++ b/dom/html/HTMLDialogElement.h	Thu Apr 29 19:35:23 2021 +0000
@@ -19,7 +19,8 @@
  public:
   explicit HTMLDialogElement(
       already_AddRefed<mozilla::dom::NodeInfo>&& aNodeInfo)
-      : nsGenericHTMLElement(std::move(aNodeInfo)) {}
+      : nsGenericHTMLElement(std::move(aNodeInfo)),
+        mPreviouslyFocusedElement(nullptr) {}
 
   NS_IMPL_FROMNODE_HTML_WITH_TAG(HTMLDialogElement, dialog)
 
@@ -58,6 +59,9 @@
  private:
   void AddToTopLayerIfNeeded();
   void RemoveFromTopLayerIfNeeded();
+  void StorePreviouslyFocusedElement();
+
+  nsWeakPtr mPreviouslyFocusedElement;
 };
 
 }  // namespace dom